core/common/cmp.hpp

```diff
@@ -5,13 +5,19 @@
 
 #pragma once
 
+#define FC_OPERATOR_NOT_EQUAL_2(L, R)              \
```

Comment:
What is a problem is resolved by this solution?
I don't think that operator != is too big or hard to hide it under the macros

Reply:
similar to 'boost::equality_comparable'